contracts/payment/TimelockedTokenEscrow.sol

```diff
@@ -0,0 +1,34 @@
+pragma solidity ^0.4.24;
+
+import "./ConditionalTokenEscrow.sol";
+
+
+/**
+ * @title TimelockedTokenEscrow
+ * @dev Escrow that holds tokens for given amount of time,
+ * preventing their whithdrawal until the time has passed.
+ */
+contract TimelockedTokenEscrow is ConditionalTokenEscrow {
+
+  uint256 public releaseTime;
+
+  /**
+   * @dev Constructor.
+   * @param _token Address of the ERC20 token that will be put in escrow.
+   * @param _releaseTime Time when the tokens will be available for withdrawal.
+   */
+  constructor (ERC20 _token, uint256 _releaseTime) public TokenEscrow(_token) {
+    // solium-disable-next-line security/no-block-members
+    require(_releaseTime > block.timestamp);
+    releaseTime = _releaseTime;
+  }
+
+  /**
+  * @dev Returns whether an address is allowed to withdraw their funds.
+  */
+  function withdrawalAllowed(address _payee) public view returns (bool) {
+    // solium-disable-next-line security/no-block-members
+    require(block.timestamp >= releaseTime);
```

Comment:
Do we want to disallow payments after this time has elapsed?

Reply:
On the one hand, if we do not disallow payments (i.e. leaving things as they are), once the time has elapsed, the escrow can at least continue working as a regular `TokenEscrow`. The owner must bare this in mind though. Should we go for this approach, we ought to explicitly describe this behavior somewhere in the contract.

On the other hand, if we do disallow payments, the contract would become useless once all tokens are withdrawn from the escrow.

Even though not entirely convinced yet, I'd go for the second option. It may be stricter, but I find it more predictable. If the owner wants a regular `TokenEscrow` after the time has elapsed, he/she should go ahead and deploy a new contract.

Reply:
Having the contract be useless after the time has elapsed doesn't worry me too much right now: this is part of a bigger issue in the ecosystem, and we haven't yet taken a stance (e.g. providing self-destruct mechanisms).

I also like the second option more, though I guess it all depends on what this sort of contract would be used for. A time-locked payment doesn't make too much sense IMO without the ability to cancel, and in all cancellable scenarios I can think of, I don't see a reason why you'd want to pay after the time has expired.